src/fmu/dataio/providers/_objectdata_base.py

```diff
@@ -0,0 +1,393 @@
+    bbox: dict = field(default_factory=dict)
+    classname: str = field(default="")
+    efolder: str = field(default="")
+    extension: str = field(default="")
+    fmt: str = field(default="")
+    layout: str = field(default="")
+    metadata: dict = field(default_factory=dict)
+    name: str = field(default="")
+    specs: dict = field(default_factory=dict)
+    subtype: str = field(default="")
+    time0: str = field(default="")
+    time1: str = field(default="")
+
+    @staticmethod
+    def _validate_get_ext(fmt: str, subtype: str, validator: dict[str, V]) -> V:
+        """Validate that fmt (file format) matches data and return legal extension."""
+        try:
+            return validator[fmt]
+        except KeyError:
+            raise ConfigurationError(
+                f"The file format {fmt} is not supported. ",
+                f"Valid {subtype} formats are: {list(validator.keys())}",
+            )
+
+    def _derive_name_stratigraphy(self) -> DerivedNamedStratigraphy:
+        """Derive the name and stratigraphy for the object; may have several sources.
+
+        If not in input settings it is tried to be inferred from the xtgeo/pandas/...
+        object. The name is then checked towards the stratigraphy list, and name is
+        replaced with official stratigraphic name if found in static metadata
+        `stratigraphy`. For example, if "TopValysar" is the model name and the actual
+        name is "Valysar Top Fm." that latter name will be used.
+        """
+        name = derive_name(self.dataio, self.obj)
+
+        # next check if usename has a "truename" and/or aliases from the config
+        strat = self.dataio.config.get("stratigraphy", {})
+        no_start_or_missing_name = strat is None or name not in strat
+
+        rv = DerivedNamedStratigraphy(
+            name=name if no_start_or_missing_name else strat[name].get("name", name),
+            alias=[] if no_start_or_missing_name else strat[name].get("alias", []),
+            stratigraphic=False
+            if no_start_or_missing_name
+            else strat[name].get("stratigraphic", False),
+            stratigraphic_alias=[]
+            if no_start_or_missing_name
+            else strat[name].get("stratigraphic_alias"),
+            offset=None if no_start_or_missing_name else strat[name].get("offset"),
+            top=None if no_start_or_missing_name else strat[name].get("top"),
+            base=None if no_start_or_missing_name else strat[name].get("base"),
+        )
+
+        if not no_start_or_missing_name and rv.name != "name":
+            rv.alias.append(name)
+
+        return rv
+
+    def _process_content(self) -> tuple[str | dict, dict | None]:
+        """Work with the `content` metadata"""
+
+        # content == "unset" is not wanted, but in case metadata has been produced while
+        # doing a preprocessing step first, and this step is re-using metadata, the
+        # check is not done.
```

Comment:
I understatt @tnatt is working at this topic now, possibly removing the 'unset' option

Reply:
I looked at it a bit with her yesterday and I think she is keeping it for the moment since it might be a breaking change. That may have changed between then and now, though

Reply:
https://github.com/equinor/fmu-dataio/issues/277